### DISCUSSION TITLE
Acentuação dados Linux

### DISCUSSION THREAD

**Discussion by adelsoncn** [HashLoad/horse#327] [General]
Criei uma API com horse acessando banco de dados Firebird 4 quando utilizo em ambiente windows o retorno dos dados a acentuação funciona perfeitamente, porém, quando gero em ambiente Linux o retorno não vem com acentuação correta.

Tentei en_US.UTF-8, pt_BR.UTF-8 utilizando os comandos:

locale-gen pt_BR.UTF-8
dpkg-reconfigure locales

**Projeto Exemplo**

https://github.com/adelsoncn/APITeste
update-locale LANG=pt_BR.UTF-8

em ambos casos a configuração é aceita no linux, porém, continua com o mesmo problema de acentuação.

A base de dados utiliza charset win1252 e collate win-ptbr, já tentei várias configurações do locate do Linux (utilizo docker/Ubuntu) e nada da certo, teria uma ideia de onde posso resolver este problema ?

**Comment by andre-djsystem** [comment]
Tente definir o charset diretamente no horse, caso use o middleware Jhonson:

Use(Jhonson('iso-8859-1'))

**Reply by adelsoncn** [reply]
ainda não deu certo infelizmente

**Reply by luisarturmenezes** [reply]
Estou com o mesmo problema porem com banco de dados SQL Server + Firedac

**Comment by dliocode** [comment]
Consegue criar um projeto com esse problema e postar aqui pra nós?

**Reply by adelsoncn** [reply]
aqui o projeto exemplo

https://github.com/adelsoncn/APITeste

**Comment by adelsoncn** [comment]
Pessoal, fiz o mesmo teste no servidor SaveInCloud e também com o mesmo problema, não sei mais o que fazer.

**Comment by cachopaweb** [comment]
Fala devs, tive esse mesmo problema, e resolvi colocando o parâmetro de CharacterSet=utf8 usando o banco de dados hospedado em ambientes linux...
   
  TFDConnection(FConnList.Items[Result]).Params.DriverID := 'FB';
  TFDConnection(FConnList.Items[Result]).Params.Database := FCaminhoBD;
  TFDConnection(FConnList.Items[Result]).Params.UserName := FUsuario;
  TFDConnection(FConnList.Items[Result]).Params.Password := FSenha;
  TFDConnection(FConnList.Items[Result]).Params.Add('CharacterSet=utf8');
  TFDConnection(FConnList.Items[Result]).Connected;

**Reply by adelsoncn** [reply]
Deu certo aqui, obrigado !

**Comment by UnisystemGIT** [comment]
Boa Tarde, no meu caso nao resolveu colcoar o  CharacterSet=utf8

**Reply by dliocode** [reply]
Consegue montar um projeto e passar para nós testarmos aqui?

**Comment by augustolpq** [comment]
Corrigi o meu trocando o characterset do fdconnection para "latin1". Uso mysql 5.7.29
